src/schemas/HederaUtils.ts

```diff
@@ -249,3 +249,51 @@ export function lookupNFTTransfer(transaction: Transaction, tokenId: string): Nf
+}
+
+export function resolveFunctionFragmentForHTSProxyContract(functionFragment: ethers.FunctionFragment, inputArgs: string): ethers.FunctionFragment {
+    let encodedFunction4BytesSignature = ""
+
+    try {
+        const inputResult = ethers.AbiCoder.defaultAbiCoder().decode(functionFragment.inputs, inputArgs)
+        encodedFunction4BytesSignature = inputResult[1].slice(0, 10)
+    } catch (failure) {
+        const f = failure as ethers.EthersError
+        if (f.code === "BUFFER_OVERRUN" || f.code === "INVALID_ARGUMENT") {
+            const inputResult = decodeRedirectForTokenInput(inputArgs)
+            encodedFunction4BytesSignature = inputResult[1].slice(0, 10)
+        }
+    }
+
+    // @notice this is the list of supported method which is collected based on HIP-218 https://hips.hedera.com/hip/hip-218
+    const ABI_FOR_SUPPORTED_METHODS = [
+        "function name() public view returns (string name)",
+        "function symbol() public view returns (string symbol)",
+        "function decimals() public view returns (uint8 decimals)",
+        "function totalSupply() external view returns (uint256 totalSupply)",
+        "function setApprovalForAll(address _operator, bool _approved) external",
+        "function ownerOf(uint256 _tokenId) external view returns (address ownerOf)",
+        "function tokenURI(uint256 _tokenId) external view returns (string tokenURI)",
+        "function balanceOf(address _owner) external view returns (uint256 balanceOf)",
+        "function balanceOf(address account) external view returns (uint256 balanceOf)",
+        "function approve(address spender, uint256 amount) external returns (bool approve)",
+        "function tokenByIndex(uint256 _index) external view returns (uint256 tokenByIndex)",
+        "function getApproved(uint256 _tokenId) external view returns (address getApproved)",
+        "function transferFrom(address _from, address _to, uint256 _tokenId) external payable",
+        "function transfer(address recipient, uint256 amount) external returns (bool transfer)",
+        "function safeTransferFrom(address _from, address _to, uint256 _tokenId) external payable",
+        "function allowance(address owner, address spender) external view returns (uint256 allowance)",
+        "function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external payable",
+        "function isApprovedForAll(address _owner, address _operator) external view returns (bool isApprovedForAll)",
+        "function transferFrom(address sender, address recipient, uint256 amount) external returns (bool transferFrom)",
+        "function tokenOfOwnerByIndex(address _owner, uint256 _index) external view returns (uint256 tokenOfOwnerByIndex)",
+      ];
+      
+    const iface = new ethers.Interface(ABI_FOR_SUPPORTED_METHODS)
+    return ethers.FunctionFragment.from({...functionFragment, outputs: iface.getFunction(encodedFunction4BytesSignature)?.outputs})
+}
```

Comment:
We could encapsulate `redirectForToken()` detection in a method and use it (two times) in `FunctionCallAnalyzer`:

```
export function isRedirectToken(contractId: string, functionHash: string): boolean {
    return contractId === HTS_PRECOMPILE_CONTRACT_ID &&
        functionHash === REDIRECT_FOR_TOKEN_FUNCTION_SIGHASH
}

```